### DISCUSSION TITLE
Can't update contract signer without page refresh

### DISCUSSION THREAD

**Discussion by emimaricic** [wevm/wagmi#1604] [Q&A]
Hey guys is there any way that I can update contract signer on useContract hook

`  const contract = useContract({
    address: process.env.NEXT_PUBLIC_RAFFLE_CONTRACT_ADDRESS,
    abi: AngleDustRaffleABI.abi,
    signerOrProvider: signer,
  });
`

**Comment by tmm** [comment]
How are you getting `signer`? It should stay up-to-date with [`useSigner`](https://wagmi.sh/react/hooks/useSigner).
